packages/web-shared/src/workflow-traces/trace-span-construction.ts

```diff
@@ -144,12 +148,17 @@ export function hookToSpan(hook: Hook, hookEvents: Event[]): Span {
 
   const lastHookReceivedEvent = hookEvents
     .slice()
-    .sort((a, b) => b.createdAt.getTime() - a.createdAt.getTime())
+    .sort(
+      (a, b) =>
+        new Date(b.createdAt).getTime() - new Date(a.createdAt).getTime()
```

Comment:
All of the code here and below is about making the UI robust against Date/string variation. This was weird before: Dates serialize to strings in JSON, but NextJS server-client boundary allows passing dates (since they do their own devaluing).
Since we're now JSON-casting all objects from the observability API, we're no longer at the mercy of whatever NextJS does, but it also means the types expect Date objects when they're actually strings.

I'll fix this more thoroughly when we start doing client-side devalue hydration.